airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/generate_erd/pipeline.py

```diff
@@ -0,0 +1,176 @@
+
+        return filtered_secrets[0]
+
+    @staticmethod
+    def _get_schema_from_discover_output(discover_output: str) -> str:
+        for line in discover_output.split("\n"):
+            json_line = json.loads(line)
+            if json_line.get("type") == "CATALOG":
+                return json.dumps(json.loads(line).get("catalog"))
+        raise ValueError("No catalog was found in output")
+
+    def _build_erd_container(self, connector_directory: Directory, discovered_catalog: str) -> Container:
+        """Create a container to run ERD generation."""
+        container = with_poetry(self.context)
+        if self.context.genai_api_key:
+            container = container.with_env_variable("GENAI_API_KEY", self.context.genai_api_key.value)
+
+        container = (
+            container.with_mounted_directory("/source", connector_directory)
+            .with_new_file("/source/erd/discovered_catalog.json", contents=discovered_catalog)
+            .with_mounted_directory("/app", self.context.erd_dir)
+            .with_workdir("/app")
+        )
+
+        return container.with_exec(["poetry", "lock", "--no-update"]).with_exec(["poetry", "install"])
+
+
+class UploadDbmlSchema(Step):
+    context: ConnectorContext
+
+    title = "Upload DBML file to dbdocs.io"
+
+    def __init__(self, context: PipelineContext) -> None:
+        super().__init__(context)
+
+    async def _run(self, connector_to_discover: Container) -> StepResult:
+        if not self.context.dbdocs_token:
+            raise ValueError(
+                "In order to publish to dbdocs, DBDOCS_TOKEN needs to be provided. Either pass the value or skip the publish step"
+            )
+
+        connector = self.context.connector
+        source_dbml_content = open(_get_dbml_file(connector.code_directory)).read()
+
+        dbdocs_container = await (
+            self.dagger_client.container()
+            .from_("node:lts-bullseye-slim")
+            .with_exec(["npm", "install", "-g", "dbdocs"])
+            .with_env_variable("DBDOCS_TOKEN", self.context.dbdocs_token.value)
+            .with_workdir("/airbyte_dbdocs")
+            .with_new_file("/airbyte_dbdocs/dbdocs.dbml", contents=source_dbml_content)
+        )
+
+        db_docs_build = ["dbdocs", "build", "dbdocs.dbml", f"--project={connector.technical_name}"]
+        await dbdocs_container.with_exec(db_docs_build).stdout()
+        # TODO: produce link to dbdocs in output logs
+
+        return StepResult(step=self, status=StepStatus.SUCCESS)
+
+
+async def run_connector_generate_erd_pipeline(
+    context: ConnectorContext,
+    semaphore: "Semaphore",
+    skip_steps: List[str],
+) -> Report:
+    context.targeted_platforms = [LOCAL_BUILD_PLATFORM]
+    steps_to_run: STEP_TREE = []
+
+    steps_to_run.append([StepToRun(id=CONNECTOR_TEST_STEP_ID.BUILD, step=BuildConnectorImages(context))])
```

Comment:
❓ I'm not sure if building the image is a requirement here, do you wan't to create ERDs for connectors in development? Creating a container from the dockerhub image and running discover would be faster.
```
connector_to_discover = context.dagger_client.container.from_(context.connector.docker_image)
```

Reply:
We indeed want to allow ERDs from dev images. Hence, I would keep the building of the image here